pkgs/tools/security/pinentry/default.nix

```diff
@@ -1,60 +1,96 @@
 { fetchurl, fetchpatch, stdenv, lib, pkgconfig, autoreconfHook
 , libgpgerror, libassuan
-, libcap ? null, libsecret ? null, ncurses ? null, gtk2 ? null, gcr ? null
-, qt4 ? null, qt5 ? null
-, enableEmacs ? false
+, ncurses, gtk2, qt
+, libcap ? null, libsecret ? null, gcr ? null
+, enabledFlavors ? [ "curses" "tty" "gtk2" "qt" "gnome3" "emacs" ]
 }:
 
-assert qt5 != null -> qt4 == null;
-assert qt4 != null -> qt5 == null;
+with stdenv.lib;
+
+assert isList enabledFlavors && enabledFlavors != [];
 
 let
   mkDerivation =
-    if qt5 != null
-      then qt5.mkDerivation
+    if (builtins.elem "qt" enabledFlavors)
+      then qt.mkDerivation
       else stdenv.mkDerivation;
+
+  mkFlag = pfxTrue: pfxFalse: cond: name:
+    "--${if cond then pfxTrue else pfxFalse}-${name}";
+  mkEnable = mkFlag "enable" "disable";
+  mkWith = mkFlag "with" "without";
+
+  mkEnablePinentry = f:
+    let
+      info = flavorInfo.${f};
+      inputs = info.buildInputs or [];
+      flag = flavorInfo.${f}.flag or null;
+      inputsSatisfied = all (f: ! isNull f) inputs;
```

Comment:
IIRC `isNull` is deprecated and we should use `!= null` instead.

Reply:
obsolete, see above comment.